src/main/flight/imu.c

```diff
@@ -342,3 +342,8 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+    return lrintf( liftOffTrust*sinf(anglerad[axis]) );
```

Comment:
This doesn't do any throttle reduction on inversion, so it doesn't suffer from the drop out I mentioned above.

https://github.com/diydrones/ardupilot/blob/e31f2d26c49d71432cb4ec16d66364bcc412703b/libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp#L9